src/Core/Injector/Injector.php

```diff
@@ -525,8 +525,8 @@ public function convertServiceProperty($value)
             return $this->get($id);
         }
 
-        // Evaluate constants surrounded by back ticks
-        if (preg_match('/^`(?<name>[^`]+)`$/', $value, $matches)) {
+        // Evaluate constants surrounded by back ticks (and quotes)
+        if (preg_match('/^[\'"]?`(?<name>[^`]+)`[\'"]?$/', $value, $matches)) {
```

Comment:
If you quote the yml string it will come out unquoted when parsed by the yml parser. You shouldn't be looking for an extra layer of quotes here.